### PR TITLE
OSIDB-3227: Duplicated CSS

### PR DESCRIPTION
**Pull request by MrMarble**
# OSIDB-3227: Duplicated CSS

## Checklist:

- [x] Commits consolidated
- [ ] Changelog updated
- [ ] Test cases added/updated
- [x] Jira ticket updated

## Summary:

Fix Bootstrap imports and reduce CSS delivery

## Changes:

While working on fixing the CSS duplication, I discovered that the bootstrap scss overrides was broken and only worked because we had duplicate code.

The variable `$theme-colors` is an special variable that is expanded by bootstrap with the `utils.scss`, this file generates all the `text-[color]` classes (among others), because we were **overwriting** the map instead of **extending** it, all this classes failed to create, for example `text-dark`, **but** because we imported bootstrap twice, in `main.ts` and `index.scss` 
the classes `text-[color]` were created by the `main.ts` import before the `index.scss` overrides.

Bootstrap overrides needs to be declared before importing bootstrap, so all the variables defined with `!default` are not created, but `maps` needs to be modified after importing `functions, variables and variables-dark`, as you can see in the docs [Add to map](https://getbootstrap.com/docs/5.3/customize/sass/#remove-from-map).

With the order of import fixed, the duplicated CSS code was removed but we were still importing `400kB` of **all** the bootstrap CSS. Since `vite` uses `postcss` under the hood, I installed [PurgeCSS](https://purgecss.com/) to remove all the CSS that is unused.

Here's a comparation of the bundle before and after all the changes (removed unrelated output):
```
vite v4.5.3 building for production...
✓ 935 modules transformed.
dist/index.html                                                0.56 kB │ gzip:   0.34 kB
dist/assets/vendor-cc057500.css                              301.27 kB │ gzip:  44.00 kB
dist/assets/index-bebddbe6.css                               333.41 kB │ gzip:  42.69 kB
dist/assets/index-d743ee3d.js                                189.85 kB │ gzip:  60.62 kB │ map: 1,441.35 kB
dist/assets/vendor-83af3c16.js                               600.45 kB │ gzip: 199.98 kB │ map: 3,109.32 kB

✓ built in 8.84s
Done in 9.15s.
```
> Before code de-duplication and purge css, 634.68kB of styles

```
vite v4.5.3 building for production...
✓ 907 modules transformed.
dist/index.html                                                0.50 kB │ gzip:   0.33 kB
dist/assets/index-5fc002f7.css                               116.54 kB │ gzip:  19.40 kB
dist/assets/index-e8726221.js                                189.85 kB │ gzip:  60.62 kB │ map: 1,441.24 kB
dist/assets/vendor-b5510954.js                               600.70 kB │ gzip: 199.80 kB │ map: 3,104.64 kB

✓ built in 8.51s
Done in 8.88s.
```
> After code de-duplication and purge css, 116.54kB of styles

## Considerations:

Since this change affects the generated code, I tried to make sure everything works as expected (modals, dropdowns, toast notifications...) but I may have left something behind, so keep a close eye after merging this (or checkout the PR locally and test)

Closes OSIDB-3327

### CODE REVIEW

**Review by C-Valen — COMMENTED** [review]
Changes look fine, will do some local tests before final approval.

+1 on purgeCSS 🔥 

Side note for the future: might be interesting to review explicit color code uses for potential possible use/extension on the theme colors schema (e.g. CvssCalculator, Affect Section Redesign, etc.).

**Review by C-Valen — APPROVED** [review]

**Comment by C-Valen** [comment]
@MrMarble Approving after a few tests looks ok.
Just probably needing a rebase to update the changelog?

**Review by superbuggy — APPROVED** [review]

**Pull request merged**
